azurelinuxagent/common/persist_firewall_rules.py

```diff
@@ -0,0 +1,235 @@
+        service_unit_file = os.path.join(self._systemd_file_path, self._network_setup_service_name)
+        try:
+            fileutil.write_file(service_unit_file,
+                                self.__SERVICE_FILE_CONTENT.format(egg_path=self._current_agent_executable_path,
+                                                                   wire_ip=self._dst_ip,
+                                                                   user_id=self._uid,
+                                                                   wait=self._wait,
+                                                                   py_path=sys.executable))
+            fileutil.chmod(service_unit_file, 0o644)
+
+            # Finally enable the service. This is needed to ensure the service is started on system boot
+            cmd = ["systemctl", "enable", self._network_setup_service_name]
+            try:
+                shellutil.run_command(cmd)
+            except CommandError as error:
+                msg = "Unable to enable service: {0}; deleting service file: {1}. Command: {2}, Exit-code: {3}.\nstdout: {4}\nstderr: {5}".format(
+                    self._network_setup_service_name, service_unit_file, ' '.join(cmd), error.returncode, error.stdout,
+                    error.stderr)
+                raise Exception(msg)
+
+        except Exception:
+            self.__remove_file_without_raising(service_unit_file)
+            raise
+
+    @staticmethod
+    def __remove_file_without_raising(file_path):
+        if os.path.exists(file_path):
+            try:
+                os.remove(file_path)
+            except Exception as error:
+                logger.warn("Unable to delete file: {0}; Error: {1}".format(file_path, ustr(error)))
+
+    def __log_network_setup_service_logs(self):
+        # Get logs from journalctl - https://www.freedesktop.org/software/systemd/man/journalctl.html
+        cmd = ["journalctl", "-u", self._network_setup_service_name, "-b"]
+        cmd_success = False
+        try:
+            stdout = shellutil.run_command(cmd)
+            msg = "Logs from the {0} since system boot: {1}".format(self._network_setup_service_name, stdout)
+            cmd_success = True
+            logger.info(msg)
+        except CommandError as error:
+            msg = "Command: {0} failed with ExitCode: {1}\nStdout: {2}\nStderr: {3}".format(' '.join(cmd),
+                                                                                            error.returncode,
+                                                                                            error.stdout, error.stderr)
+            logger.warn(msg)
+        except Exception as error:
+            msg = "Ran into unexpected error when getting logs for {0} service. Error: {1}".format(
+                self._network_setup_service_name, ustr(error))
+            logger.warn(msg)
+
+        add_event(
+            op=WALAEventOperation.PersistFirewallRules,
+            is_success=cmd_success,
```

Comment:
but this is the status of the journalctl command; we should get the status of the service

Reply:
This is actually the logs of the service (the same logs that you see when we do systemctl status <service>). It would get the verbose logs of the service but would probably not get the status of it (active, failed, etc). I was initially considering that but then I figured it would be more helpful to get the complete logs than the status. Do you think status makes more sense?

Reply:
Btw this is an example of how journalctl logs look like - 

```log
2021-02-05T06:57:44.194001Z INFO ExtHandler ExtHandler Logs from the walinuxagent-network-setup.service since system boot: -- 
Logs begin at Fri 2021-02-05 06:57:36 UTC, end at Fri 2021-02-05 06:57:44 UTC. --
Feb 05 06:57:38 edpwz5814j python3[441]: Setting up firewall for the WALinux Agent with args: {'wait': '-w', 'uid': '0', 'dst_ip': '168.63.129.16'}
Feb 05 06:57:38 edpwz5814j systemd[1]: Started Setup network rules for WALinuxAgent.
```

Reply:
I meant the is_success we send in telemetry. Probably we do not care about failures executing journalctl, but we do care if the firewall rules were not created

Reply:
Ahh good point, let me also query the service to see if it was in a failed state or not and send the is-success based on that